queries/javascript/highlights.scm

```diff
@@ -0,0 +1,181 @@
+
+(function
+  name: (identifier) @function)
+(function_declaration
+  name: (identifier) @function)
+(method_definition
+  name: (property_identifier) @function.method)
+
+(pair
+  key: (property_identifier) @function.method
+  value: (function))
+(pair
+  key: (property_identifier) @function.method
+  value: (arrow_function))
+
+(assignment_expression
+  left: (member_expression
+    property: (property_identifier) @function.method)
+  right: (arrow_function))
+(assignment_expression
+  left: (member_expression
+    property: (property_identifier) @function.method)
+  right: (function))
+
+(variable_declarator
+  name: (identifier) @function
+  value: (arrow_function))
+(variable_declarator
+  name: (identifier) @function
+  value: (function))
+
+(assignment_expression
+  left: (identifier) @function
+  right: (arrow_function))
+(assignment_expression
+  left: (identifier) @function
+  right: (function))
+
+; Function and method calls
+;--------------------------
+
+(call_expression
+  function: (identifier) @function)
+
+(call_expression
+  function: (member_expression
+    property: (property_identifier) @function.method))
+
+; Variables
+;----------
+
+(formal_parameters (identifier) @variable.parameter)
+
+(identifier) @variable
+
+; Properties
+;-----------
+
+(property_identifier) @property
+
+; Literals
+;---------
+
+(this) @variable.builtin
+(super) @variable.builtin
+
+(true) @constant.builtin
+(false) @constant.builtin
+(null) @constant.builtin
+(comment) @comment
+(string) @string
+(regex) @string.special
+(template_string) @string
+(number) @number
+
+; Punctuation
+;------------
+
+(template_substitution
+  "${" @punctuation.special
+  "}" @punctuation.special) @embedded
+
+";" @punctuation.delimiter
+"." @punctuation.delimiter
+"," @punctuation.delimiter
+
+"--" @operator
+"-" @operator
+"-=" @operator
+"&&" @operator
+"+" @operator
+"++" @operator
+"+=" @operator
+"<" @operator
+"<<" @operator
+"=" @operator
+"==" @operator
+"===" @operator
+"=>" @operator
+">" @operator
+">>" @operator
+"||" @operator
+
+"(" @punctuation.bracket
+")" @punctuation.bracket
+"[" @punctuation.bracket
+"]" @punctuation.bracket
+"{" @punctuation.bracket
+"}" @punctuation.bracket
+
+; Keywords
+;----------
+
+"as" @keyword
+"async" @keyword
+"await" @keyword
+"break" @keyword
+"case" @keyword
+"catch" @keyword
+"class" @keyword
+"const" @keyword
+"continue" @keyword
+"debugger" @keyword
+"default" @keyword
+"delete" @keyword
+"do" @keyword
+"else" @keyword
+"export" @keyword
+"extends" @keyword
+"finally" @keyword
+"for" @keyword
+"from" @keyword
+"function" @keyword
+"get" @keyword
+"if" @keyword
+"import" @keyword
+"in" @keyword
+"instanceof" @keyword
+"let" @keyword
+"new" @keyword
+"of" @keyword
+"return" @keyword
+"set" @keyword
+"static" @keyword
+"switch" @keyword
+"target" @keyword
+"throw" @keyword
+"try" @keyword
+"typeof" @keyword
+"var" @keyword
+"void" @keyword
+"while" @keyword
+"with" @keyword
+"yield" @keyword
```

Comment:
Everything is factorable into one group like this :
```
[
"else",
...
] @keyword
```

Furthermore it might be also good to split those keywords depending on their type (like `@conditionnal` and such). please refer to the `CONTRIBUTING.md` for more informations on that